tests/Microsoft.Identity.Test.Unit/BrokerTests/BrokerRequestTests.cs

```diff
@@ -290,6 +295,128 @@ public void GetAccountWithDuplicateBrokerAccountsTest()
             Assert.AreEqual(TestConstants.HomeAccountId, account.HomeAccountId.Identifier);
         }
 
+        [TestMethod]
+        public void CreateFromiOSBroker_HandlesAllErrorFields()
+        {
+            // Arrange
+            Dictionary<string, string> responseDictionary = new Dictionary<string, string>();
+            responseDictionary["error_metadata"] = @"{""home_account_id"":""test_home"", ""username"" : ""Testacct@rreretr.com"" }";
+            responseDictionary[BrokerResponseConst.BrokerErrorCode] = TestErrCode;
+            responseDictionary[OAuth2ResponseBaseClaim.SubError] = TestSuberrCode;
+            responseDictionary[BrokerResponseConst.BrokerErrorDescription] = TestErrDescr;
+
+            // act
+            var token = MsalTokenResponse.CreateFromiOSBrokerResponse(responseDictionary);
+
+            // assert
+            Assert.AreEqual(TestErrCode, token.Error);
+            Assert.AreEqual(TestSuberrCode, token.SubError);
+            Assert.AreEqual(TestErrDescr, token.ErrorDescription);
+            Assert.AreEqual("test_home", token.AccountUserId);
+            Assert.AreEqual("Testacct@rreretr.com", token.Upn);
```

Comment:
I think we may already have some of the constants used in these tests in the TestConstants class. Maybe see what you can get from there.